modules/upsell-order-bump/includes/OrderBump.php

```
@@ -121,4 +159,122 @@ public function woocommerce_custom_price_to_cart_item( $cart_object ) {
+				}
+			} else {
+				$cart_categories = wp_get_post_terms( $cart_product_id, 'product_cat', array( 'fields' => 'ids' ) );
+				$target_categories = $bump_info['target_categories'];
+				if ( ! empty( array_intersect( $cart_categories, $target_categories ) ) ) {
+					$remaining_targets = array_merge( $remaining_targets, $cart_item_ids );
+				}
+			}
+		}
+
+		return $remaining_targets;
+	}
```

PHPMD (2.15.0)

Avoid unused private methods such as 'get_remaining_target_products'. (Unused Code Rules)

(UnusedPrivateMethod)